src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    this.forEach((element, index) => {
+      let elementToAdd = element;
+      let separatorToAdd = separator;
+
+      if (separator === null) {
+        separatorToAdd = 'null';
+      }
```

Comment:
`separator` won't change during the function execution, so there is no need to check it on every step, you may do it only once before your `forEach`